### PR TITLE
Metrics for cluster health status

### PR DESCRIPTION
**Pull request by kmjayadeep**
* Integrate metrics framework
* Create new gauge metrics for exposing cluster health status
* Reduce reconcile interval in test cases to speed up tests.

### CODE REVIEW

**Review by richiesebastian — APPROVED** [review]

**Review by PuranAdhikari — APPROVED** [review]
LGTM.

**Review by gourishkb — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/kubeslice/apis v0.1.3
 	github.com/kubeslice/gateway-sidecar v0.2.0
-	github.com/kubeslice/kubeslice-monitoring v0.1.6
+	github.com/kubeslice/kubeslice-monitoring v0.1.7-0.20230404135238-9338c8ec16f9
```
Comment (gourishkb) [inline go.mod:14]: nit: can we release a tag from monitoring repo and use it here ?

Comment (kmjayadeep) [reply go.mod:14]: yes. i will do it in the next pr

**Pull request merged**